### PR TITLE
Upgraded ThriftRW to include optimizations

### PR DESCRIPTION
**Pull request by alexeykudinkin**
More details in:
https://github.com/thriftrw/thriftrw-node/pull/184

### CODE REVIEW

File: benchmarks/multi_bench.js [hunk 1]
```diff
@@ -28,6 +28,10 @@ var setTimeout = require('timers').setTimeout;
 var Buffer = require('buffer').Buffer;
 var console = require('console');
 
+// Node.js deprecated Buffer in favor of Buffer.alloc and Buffer.from.
+// istanbul ignore next
+var bufferFrom = Buffer.from || Buffer;
```
Comment (srivastavankit) [inline benchmarks/multi_bench.js:33]: Maybe move all the `Buffer.alloc` & `Buffer.from` to a common file?

File: as/thrift.js [hunk 2]
```diff
@@ -54,13 +54,10 @@ function TChannelAsThrift(opts) {
         allowIncludeAlias: opts.allowIncludeAlias,
         allowOptionalArguments: opts.allowOptionalArguments,
         defaultAsUndefined: opts.defaultAsUndefined,
+        releaseSources: true,
```
Comment (srivastavankit) [inline as/thrift.js:57]: maybe add a one line comment on what this flag does?

File: as/thrift.js [hunk 3]
```diff
@@ -148,11 +145,11 @@ function registerMeta(metaSource) {
     var self = this;
 
     var metaSpec = new thriftrw.Thrift({
-        source: metaSource
+        source: metaSource,
+        releaseSources: true
```
Comment (srivastavankit) [inline as/thrift.js:149]: same as above?

**Review by srivastavankit — APPROVED** [review]

File: as/thrift.js (hunk 2 above)
Comment (alexeykudinkin) [reply as/thrift.js:57]: SG

File: as/thrift.js (hunk 3 above)
Comment (alexeykudinkin) [reply as/thrift.js:149]: SG

**Pull request merged**